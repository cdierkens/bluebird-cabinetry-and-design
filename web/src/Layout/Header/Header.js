import { Link } from "gatsby";
import React from "react";
import resolveConfig from "tailwindcss/resolveConfig";
import tailwindConfig from "../../../tailwind.config.js";
import { useMatchMedia } from "../../hooks";
import DesktopNav from "./DesktopNav/DesktopNav.js";
import styles from "./Header.module.css";
import MobileNav from "./MobileNav/MobileNav.js";

const { theme } = resolveConfig(tailwindConfig);

const Links = () => (
  <>
    <Link className={styles.Link} to="/">
      Home
    </Link>
    <Link className={styles.Link} to="/about">
      About
    </Link>
    <Link className={styles.Link} to="/services">
      Services
    </Link>
    <Link className={styles.Link} to="/portfolio">
      Portfolio
    </Link>
    <Link className={styles.Button} to="/contact">
      Contact
    </Link>
  </>
);

const Header = () => {
  const isMobile = useMatchMedia(`(max-width: ${theme.screens.md})`);

  return (
    <header className={styles.Header}>
      {isMobile ? (
        <MobileNav>
          <Links />
        </MobileNav>
      ) : (
        <DesktopNav>
          <Links />
        </DesktopNav>
      )}
    </header>
  );
};

export default Header;
