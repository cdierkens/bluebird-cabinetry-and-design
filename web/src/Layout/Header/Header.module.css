.Header {
  @apply bg-transparent-white fixed w-full z-10;
}

.Nav {
  @apply max-w-screen-lg w-full mx-auto grid grid-cols-8 gap-3;
}

.Logo {
  @apply bg-blue-dark rounded-b-md p-9 w-48 absolute z-10;
}

.Main {
  @apply col-span-6 flex justify-between items-center;
}

.NavLink {
  @apply font-sans font-bold tracking-tight text-blue-dark p-4 my-4 mx-5 text-xl;
}

.Button {
  @apply font-body font-semibold rounded-md border border-blue-dark text-blue-dark py-3 px-9 text-xl;
}

.LogoWrapper {
  @apply col-span-2 relative transform origin-top transition duration-500 scale-100;
}

.LogoWrapperTransform {
  @apply scale-50;
}
