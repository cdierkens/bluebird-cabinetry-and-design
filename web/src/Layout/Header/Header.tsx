import { Link } from "gatsby";
import React from "react";
import { useMatchMedia } from "../../hooks";
import { theme } from "../../lib/tailwind.js";
import DesktopNav from "./DesktopNav";
import * as styles from "./Header.module.css";
import MobileNav from "./MobileNav";

const Links: React.FC = () => (
  <>
    <Link className={styles.Link} activeClassName={styles.LinkActive} to="/">
      Home
    </Link>
    <Link
      className={styles.Link}
      activeClassName={styles.LinkActive}
      to="/about"
    >
      About
    </Link>
    <Link
      className={styles.Link}
      activeClassName={styles.LinkActive}
      to="/designers"
    >
      Designers
    </Link>
    <Link
      className={styles.Link}
      activeClassName={styles.LinkActive}
      to="/services"
    >
      Services
    </Link>
    <Link
      className={styles.Link}
      activeClassName={styles.LinkActive}
      to="/portfolio"
    >
      Portfolio
    </Link>
    <Link
      className={styles.Link}
      activeClassName={styles.LinkActive}
      to="/contact"
    >
      Contact
    </Link>
  </>
);

const Header: React.FC = () => {
  const isMobile = useMatchMedia(`(max-width: ${theme.screens.md})`);

  return (
    <header className={styles.Header}>
      {isMobile ? (
        <MobileNav>
          <Links />
        </MobileNav>
      ) : (
        <DesktopNav>
          <Links />
        </DesktopNav>
      )}
    </header>
  );
};

export default Header;
