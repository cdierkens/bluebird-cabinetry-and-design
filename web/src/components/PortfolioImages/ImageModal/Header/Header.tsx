import React from "react";
import { todo } from "../../../../migration.types";

const Header: React.FC<todo> = ({
  currentView: { title },
  modalProps: { onClose, toggleFullscreen, isFullscreen },
}) => {
  if (isFullscreen) {
    return (
      <div>
        {title && (
          <div className="inline absolute z-10 top-6 left-2 text-left px-2 py-1 text-sm rounded bg-black text-gray-light bg-opacity-25">
            <span className="px-1 whitespace-nowrap truncate">{title}</span>
          </div>
        )}

        <div>
          <button
            onClick={toggleFullscreen}
            className="fixed z-50 top-0 right-0 inline-block p-4 m-2 text-gray-light hover:text-gold focus:outline-none focus:ring bg-black bg-opacity-25 rounded"
          >
            <span className="sr-only">Close</span>
            <svg
              role="presentation"
              viewBox="0 0 24 24"
              className="h-6 w-6 fill-current stroke-current"
            >
              <path d="M18.984 6.422l-5.578 5.578 5.578 5.578-1.406 1.406-5.578-5.578-5.578 5.578-1.406-1.406 5.578-5.578-5.578-5.578 1.406-1.406 5.578 5.578 5.578-5.578z"></path>
            </svg>
          </button>
        </div>
      </div>
    );
  }

  return (
    <div className="p-1 md:p-4 bg-gray-light shadow-sm flex justify-between items-center">
      <div className="truncate" style={{ maxWidth: "calc(100% - 80px)" }}>
        {title && (
          <span className="px-1 whitespace-nowrap truncate">{title}</span>
        )}
      </div>
      <div>
        <button
          onClick={toggleFullscreen}
          className="inline-block p-2 hover:text-black text-gray-darker focus:outline-none focus:ring"
        >
          <span className="sr-only">Fullscreen</span>
          <svg
            className="h-6 w-6 fill-current stroke-current"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path d="M14.016 5.016h4.969v4.969h-1.969v-3h-3v-1.969zM17.016 17.016v-3h1.969v4.969h-4.969v-1.969h3zM5.016 9.984v-4.969h4.969v1.969h-3v3h-1.969zM6.984 14.016v3h3v1.969h-4.969v-4.969h1.969z"></path>
          </svg>
        </button>
        <button
          onClick={onClose}
          className="inline-block p-2 hover:text-black text-gray-darker focus:outline-none focus:ring"
        >
          <span className="sr-only">Close</span>
          <svg
            role="presentation"
            viewBox="0 0 24 24"
            className="h-6 w-6 fill-current stroke-current"
          >
            <path d="M18.984 6.422l-5.578 5.578 5.578 5.578-1.406 1.406-5.578-5.578-5.578 5.578-1.406-1.406 5.578-5.578-5.578-5.578 1.406-1.406 5.578 5.578 5.578-5.578z"></path>
          </svg>
        </button>
      </div>
    </div>
  );
};

export default Header;
