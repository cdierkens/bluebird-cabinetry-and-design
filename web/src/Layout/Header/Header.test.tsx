import { render } from "@testing-library/react";
import React from "react";
import Header from "./Header";

it("matches snapshot", () => {
  const { asFragment } = render(<Header />);

  expect(asFragment()).toMatchSnapshot();
});
