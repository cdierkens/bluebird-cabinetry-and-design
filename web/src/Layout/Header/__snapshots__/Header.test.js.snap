// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <nav
      class="DesktopNav grid-cols-6 lg:grid-cols-8"
    >
      <div
        class="HeaderLogo col-span-1 lg:col-span-2"
      >
        <a
          href="/"
        >
          <img
            alt="Bluebird Logo"
            class="Logo "
            src="test-file-stub"
          />
        </a>
      </div>
      <div
        class="HeaderNav col-span-5 lg:col-span-6"
      >
        <a
          class="Link"
          href="/"
        >
          Home
        </a>
        <a
          class="Link"
          href="/about"
        >
          About
        </a>
        <a
          class="Link"
          href="/services"
        >
          Services
        </a>
        <a
          class="Link"
          href="/portfolio"
        >
          Portfolio
        </a>
        <a
          class="ButtonBlue Button"
          href="/contact"
        >
          Contact
        </a>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
